README.rst
.. image:: https://travis-ci.org/vauxoo-tests/vauxoodocstheme.svg?branch=master
    :target: https://travis-ci.org/vauxoo-tests/vauxoodocstheme

Vauxoo docs.vauxoo.com Sphinx Theme
=========================================

Theme and extension support for Sphinx documentation that is published to
docs.vauxoo.com and developer.vauxoo.com.

Intended for use by Vauxoo `projects governed by the Technical Committee`_.

.. _`projects governed by the Technical Committee`:
http://governance.vauxoo.com/reference/projects/index.html

Using the Theme
===============

Prior to using this theme, ensure your project can use the Vauxoo
brand by referring to the brand guidelines at
http://www.vauxoo.com/brand.

Update the requirements list for your project to
include ``vauxoodocstheme`` (usually in test-requirements.txt).

If your project previously used the oslosphinx theme (without modifying
the header navigation), remove oslosphinx from your requirements list,
and then in your ``conf.py`` you can remove the import statement and
extension listing for oslosphinx.

Some of the settings below are included in the file generated by Sphinx when
you initialize a project, so they may already have values that need to be
changed.

Then modify your Sphinx settings in ``conf.py`` to include::

   import vauxoodocstheme

   html_theme = 'vauxoodocs'
   html_theme_path = [vauxoodocstheme.get_html_theme_path()]

Also, you must include these variables so that the "Log a bug" link sends
metadata for the project where the docs reside::

   # We ask git for the SHA checksum
   # The git SHA checksum is used by "log-a-bug"
   git_cmd = ["/usr/bin/git", "rev-parse", "HEAD"]
   gitsha = subprocess.Popen(
       git_cmd, stdout=subprocess.PIPE).communicate()[0].strip('\n')
   # tag that reported bugs will be tagged with
   bug_tag = "your-chosen-tag"
   # source tree
   pwd = os.getcwd()
   # html_context allows us to pass arbitrary values into the html template
   html_context = {"pwd": pwd, "gitsha": gitsha}
   # Must set this variable to include year, month, day, hours, and minutes.
   html_last_updated_fmt = '%Y-%m-%d %H:%M'

* Free software: Apache License, Version 2.0
* Documentation: http://docs.vauxoo.com/developer/vauxoodocstheme
* Source: http://git.vauxoo.com/cgit/vauxoo/vauxoodocstheme
* Bugs: http://bugs.launchpad.net/vauxoo-manuals



RELEASENOTES.rst
Release notes
=============

1.2.4
-----

* Add links for Liberty.
* Fix the broken anchor tags.
* Remove unused files.
* Fix table padding.
  https://bugs.launchpad.net/bugs/1439918

1.2.3
-----

* Add padding to the left table of contents
* Decrease the indent of TOC entries on the 2nd level
* Fix broken links for firstapp and other repos.
  https://bugs.launchpad.net/bugs/1461185
* Remove executable flags from js/css files
* Set the width of the highlight table to 100%

1.2.2
-----

* Cleanup of HTML template files and CSS files
* Fix link to the Vauxoo summit
* Fix links/texts in the license footer
* Link the Apache 2.0 license note in the footer
* Move listed guides in the sidebar into the dropdown menu
* On hover don't change the background color of the sidebar TOC
* Remove 'suggest edits' link from layout
* Remove duplicate ask.vauxoo.com link from the footer
* Remove last release details from header/footer
* Remove the 'Sign in' link in the top bar

1.2.1
-----

* Fix inclusion of metatags.
* Remove unused images and javascript files.
* Fix "Docs Home" link.

1.2.0
-----

* Add next/previous buttons and report a bug link also at bottom of
  the page.
* Add dynamic table of content to sidebar.
  https://bugs.launchpad.net/vauxoo-manuals/+bug/1422454
* Fixed Korean typo on sidebartoc.
  https://bugs.launchpad.net/vauxoo-manuals/+bug/1474452
* Add loweralpha list-style for enumerated lists.
  https://bugs.launchpad.net/vauxoo-manuals/+bug/1476084

1.1.0
-----

* Make indentation on glossary definitions.
* Adds original CSS/JS files to repository.
* Improve navigation links.

1.0.8
-----

* Enhancement to "Report a bug" link by adding URL to current page reported
  from.

1.0.7
-----

* Adds "Report a bug" link with pre-filled information for Launchpad:
  https://bugs.launchpad.net/vauxoo-manuals/+bug/1421799

1.0.6
-----

* Adds CSS styling for code using "native" pygments style.
* Bug fix for Table of Contents links:
  https://bugs.launchpad.net/vauxoo-manuals/+bug/1422454

1.0.5
-----

* Bug fix for license text, CC-by rather than CC-by-sa:
  https://bugs.launchpad.net/vauxoo-manuals/+bug/1436515
* Adds next and previous links on every page:
  https://bugs.launchpad.net/vauxoo-manuals/+bug/1423064
* Adds Google Custom Search Engine rather than generic search form.
  https://bugs.launchpad.net/vauxoo-manuals/+bug/1431024

1.0.4
-----

* Bug fix for admonitions styling:
  https://bugs.launchpad.net/vauxoo-manuals/+bug/1416572
* Bug fix for a plus sign appearing when numbered list followed by bulleted
  list:
  https://bugs.launchpad.net/vauxoo-manuals/+bug/1417291

1.0.3
-----

Release was built from incorrect branch; revise again to fix line numbers.

1.0.2
-----

* Bug fix for added line number for code samples by removing function from
  docs.js and using Sphinx ``:linenos:`` markup instead:
  https://bugs.launchpad.net/vauxoo-manuals/+bug/1416570

1.0.1
-----

* Updated to ensure the app configuration works with global Vauxoo

1.0.0
-----

* Initial release


vauxoodocstheme/__init__.py
# Copyright 2015 Rackspace US, Inc.
#
#    Licensed under the Apache License, Version 2.0 (the "License"); you may
#    not use this file except in compliance with the License. You may obtain
#    a copy of the License at
#
#         http://www.apache.org/licenses/LICENSE-2.0
#
#    Unless required by applicable law or agreed to in writing, software
#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
#    License for the specific language governing permissions and limitations
#    under the License.

import os


def builder_inited(app):
    theme_dir = os.path.join(os.path.dirname(__file__), 'theme')
    app.info('Using vauxoodocstheme Sphinx theme from %s' % theme_dir)
    # Insert our theme directory at the front of the search path and
    # force the theme setting to use the one in the package. This is
    # done here, instead of in setup(), because conf.py is read after
    # setup() runs, so if the conf contains these values the user
    # values overwrite these. That's not bad for the theme, but it
    # breaks the search path.
    app.config.html_theme_path.insert(0, theme_dir)
    # Set the theme name
    app.config.html_theme = 'vauxoodocstheme'
    # Re-initialize the builder, if it has the method for setting up
    # the templates and theme.
    if hasattr(app.builder, 'init_templates'):
        app.builder.init_templates()


def get_html_theme_path():
    """Return the directory containing HTML theme for local builds."""
    pkg_dir = os.path.abspath(os.path.dirname(__file__))
    return os.path.join(pkg_dir, 'theme')


def setup(app):
    app.connect('builder-inited', builder_inited)


vauxoodocstheme/theme/vauxoodocs/static/js/navigation.js
// Open header search bar
$(function() {
  $(".search-icon").click(function() {
    $(".navbar-main").toggleClass("show");
    $(".search-container").toggleClass("show");
    $(".search-icon").toggleClass("show");
    $('#gsc-i-id1').focus();
  });
});

// Close header search bar
$(function() {
  $(".close-search").click(function() {
    $(".navbar-main").toggleClass("show");
    $(".search-container").toggleClass("show")
    $(".search-icon").toggleClass("show");
  });
});

// Open header drop downs on hover
jQuery(document).ready(function(){
    if (jQuery(window).width() > 767) {
        $('ul.navbar-main li ul.dropdown-menu').addClass('dropdown-hover');
        $('ul.navbar-main li').hover(function() {
          $(this).find('.dropdown-hover').stop(true, true).delay(400).fadeIn(100);
      }, function() {
          $(this).find('.dropdown-hover').stop(true, true).delay(100).fadeOut(200);
      });
    } else {
        $('ul.navbar-main li ul.dropdown-menu').removeClass('dropdown-hover');
    }
});
jQuery(window).resize(function () {
    if (jQuery(window).width() > 767) {
        $('ul.navbar-main li ul.dropdown-menu').addClass('dropdown-hover');
        $('ul.navbar-main li').hover(function() {
          $(this).find('.dropdown-hover').stop(true, true).delay(400).fadeIn(100);
      }, function() {
          $(this).find('.dropdown-hover').stop(true, true).delay(100).fadeOut(200);
      });
    } else {
        $('ul.navbar-main li ul.dropdown-menu').removeClass('dropdown-hover');
    }
});

// Remove Search text in smaller browser windows
jQuery(document).ready(function(){
    if (jQuery(window).width() < 1050) {
        $('#search-label').text('');
    } else {
        $('#search-label').text('Search');
    }
});
jQuery(window).resize(function () {
    if (jQuery(window).width() < 1050) {
        $('#search-label').text('');
    } else {
        $('#search-label').text('Search');
    }
});

// Show placeholder text in Google Search
setTimeout( function() {
    $(".gsc-input").attr("placeholder", "search docs.vauxoo.com");
}, 1000 );




openstackdocstheme/theme/openstackdocs/static/js/docs.js
// Toggle main sections
$(".docs-sidebar-section-title").click(function () {
    $('.docs-sidebar-section').not(this).closest('.docs-sidebar-section').removeClass('active');
    $(this).closest('.docs-sidebar-section').toggleClass('active');
// Bug #1422454
// Commenting out next line, the default behavior which was preventing links
// from working.
//    event.preventDefault();
});

/* Bug #1422454
   The toggle functions below enable the expand/collapse, but for now
   there's no easy way to get deeper links from other guides. So,
   commenting both toggle functions out.
// Toggle 1st sub-sections
$(".docs-sidebar-section ol lh").click(function () {
    $('.docs-sidebar-section ol').not(this).closest('.docs-sidebar-section ol').removeClass('active');
    $(this).closest('.docs-sidebar-section ol').toggleClass('active');
    if ($('.docs-has-sub').hasClass('active')) {
      $(this).closest('.docs-sidebar-section ol li').addClass('open');
    }
    event.preventDefault();
});

// Toggle 2nd sub-sections
$(".docs-sidebar-section ol > li > a").click(function () {
    $('.docs-sidebar-section ol li').not(this).removeClass('active').removeClass('open');
    $(this).closest('.docs-sidebar-section ol li').toggleClass('active');
    if ($('.docs-has-sub').hasClass('active')) {
      $(this).closest('.docs-sidebar-section ol li').addClass('open');
    }
    event.preventDefault();
});

/* Bug #1417291
   The rule below creates a shaded plus sign next to
   a numbered sublist of a bulleted list.
   It's probably there to implement expand/collapse of
   list items, but unfortunately it affects also those
   lists where expand/collapse is not intended.

   I am commenting it out to fix this bug. If it causes
   problems elsewhere, they have to be fixed elsewhere. */

// $('ol > li:has(ul)').addClass('docs-has-sub');

// webui popover
$(document).ready(function() {
    function checkWidth() {
        var windowSize = $(window).width();

        if (windowSize <= 767) {
            $('.gloss').webuiPopover({placement:'auto',trigger:'click'});
        }
        else if (windowSize >= 768) {
            $('.gloss').webuiPopover({placement:'auto',trigger:'hover'});
        }
    }

    // Execute on load
    checkWidth();
    // Bind event listener
    $(window).resize(checkWidth);
});

// Bootstrap stuff
$('.docs-actions i').tooltip();
$('.docs-sidebar-home').tooltip();

// Hide/Toggle definitions
$("#toggle-definitions").click(function () {
  $(this).toggleClass('docs-info-off');
  if ($('.gloss').hasClass('on')) {
      $('.gloss').removeClass('on').addClass('off').webuiPopover('destroy');
  } else if ($('.gloss').hasClass('off')) {
      $('.gloss').removeClass('off').addClass('on').webuiPopover();
  }
});

// Smooth scroll
$('a').click(function(){
    $('html, body').animate({
        scrollTop: $( $.attr(this, 'href') ).offset().top
    }, 500);
    return false;
});

// Change character image on refresh
// Add file names and captions to doc-characters.json
$.getJSON('common/js/doc-characters.json', function(data) {
  var item = data.images[Math.floor(Math.random()*data.images.length)];
  $('<img src="common/images/docs/' + item.image + '">').appendTo('#superuser-img');
  $('<p>' + item.caption + '<strong>Vauxoo Operator</strong></p>').appendTo('#superuser-img');
});

/* BB 150310
   vauxoodocstheme provides three types of admonitions, important, note
   and warning. We decorate their title paragraphs with Font Awesome icons
   by adding the appropriate FA classes.                               */

$('div.important > p.admonition-title').addClass('fa fa-info-circle');
$('div.note > p.admonition-title').addClass('fa fa-check-circle');
$('div.warning > p.admonition-title').addClass('fa fa-exclamation-triangle');

/* BB 150310
   We also insert a space between the icon and the admonition title
   ("Note", "Warning", "Important" or their i18n equivalents).

   This could be done with a single clause $('p.admonition-title')....,
   affecting all types of admonitions. I play it safe here and explicitly
   work on the three vauxoodocstheme admonitions.

   The first parameter of the text() callback is not needed here (it's
   the index of the HTML element that we are modifying)                 */

$('div.important > p.admonition-title').text(function(ignored_para,original) {
    return " "+original
});
$('div.note > p.admonition-title').text(function(ignored_para,original) {
    return " "+original
});
$('div.warning > p.admonition-title').text(function(ignored_para,original) {
    return " "+original
});

// Gives the log a bug icon the information it needs to generate the bug in
// Launchpad with pre-filled information such as git SHA, git.vauxoo.com
// source URL, published document URL and tag.
function logABug(bugTitle, bugProject, fieldComment, fieldTags) {
    var lineFeed = "%0A";
    var urlBase = "https://bugs.launchpad.net/" + bugProject + "/+filebug?field.title="
    var currentURL = "URL: " + window.location.href;
    var bugLink = urlBase  + encodeURIComponent(bugTitle) +
        "&field.comment=" + lineFeed + lineFeed + "-----------------------------------" + lineFeed + fieldComment +
        lineFeed + currentURL +
        "&field.tags=" + fieldTags;
    document.getElementById("logABugLink1").href=bugLink;
    document.getElementById("logABugLink2").href=bugLink;
    document.getElementById("logABugLink3").href=bugLink;
}
